src/c_api/c_predict_api.cc

```diff
@@ -210,18 +210,26 @@ int _CreatePartialOut(const char* symbol_json_str,
 
   std::vector<NDArray> arg_arrays, aux_arrays;
   for (size_t i = 0; i < arg_shapes.size(); ++i) {
-    NDArray nd = NDArray(arg_shapes[i], ctx);
     if (arg_params.count(arg_names[i]) != 0) {
+      int dtype = arg_params[arg_names[i]].dtype();
+      NDArray nd = NDArray(arg_shapes[i], ctx, false, dtype);
       CopyFromTo(arg_params[arg_names[i]], &nd);
+      arg_arrays.push_back(nd);
+    } else {
+      NDArray nd = NDArray(arg_shapes[i], ctx);
+      arg_arrays.push_back(nd);
```

Comment:
can this line be moved to after the if-else instead of adding in if separately and else separately?